### ISSUE TITLE
Control flow schematic: ngIf with else condition template not migrated correctly

### ISSUE THREAD

**Issue by mattlewis92** [angular/angular#53361]
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

No

### Description

Sorry, it looks like I missed a few straggling issues! 😅 

Input (`defaultTemplate` is assumed to be a template passed in via a component input):
```
<button *ngIf="active; else defaultTemplate">Hello!</button> 
```

Output:
```
@if (active) {
  <button>Hello!</button>
} @else {
  #defaultTemplate|
}
```

Expected output:
```
@if (active) {
  <button>Hello!</button>
} @else {
  <ng-template [ngTemplateOutlet]="defaultTemplate"></ng-template>
}

```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 17.0.5
Node: 20.9.0
Package Manager: pnpm 8.11.0
OS: darwin arm64

Angular: 17.0.5+sha-2d2eaec
... core

Package                             Version
-------------------------------------------------------------
@angular-devkit/architect           0.1700.1
@angular-devkit/build-angular       17.0.5
@angular-devkit/core                17.0.5
@angular-devkit/schematics          17.0.5
@angular/animations                 17.0.5
@angular/cdk                        17.0.1
@angular/cli                        17.0.5
@angular/common                     17.0.5
@angular/compiler                   17.0.5
@angular/compiler-cli               17.0.5
@angular/elements                   17.0.5
@angular/forms                      17.0.5
@angular/google-maps                15.2.9
@angular/language-service           17.0.5
@angular/localize                   17.0.5
@angular/platform-browser           17.0.5
@angular/platform-browser-dynamic   17.0.5
@angular/router                     17.0.5
@angular/service-worker             17.0.5
@schematics/angular                 17.0.5
ng-packagr                          17.0.2
rxjs                                7.8.0
typescript                          5.2.2
webpack                             5.88.2
zone.js                             0.14.2
```

### Anything else?

This is using the latest core-builds version via `"@angular/core": "angular/core-builds#17.0.5+2d2eaec",`

**Comment by jessicajaniuk** [comment]
Ah, yeah this wouldn't migrate properly because the ng-template does not exist in the same component template. I think I can handle this case.

**Comment by jessicajaniuk** [comment]
Fix is in #53368. Thanks!
